dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+            sequences = new HashMap<String, String>();
+            sequences.put("permission", "permission_seq");
+            sequences.put("users_to_delete", "user_to_delete_seq");
+            tableIDColumns = new HashMap<String, String>();
+            tableIDColumns.put("permission", "id");
+            tableIDColumns.put("users_to_delete", "id");
+
+        }
+    }
+
+    /**
+     * Takes a ZIP file from the /temp directory to restore dotCMS data. Currently, it will blow away
+     * all current data. This method cannot currently be run in a transaction. For performance reasons
+     * with DB drivers and connections it closes the session every so often.
+     *
+     * @param out - A print writer for output.
+     * @throws IOException
+     * @throws Exception
+     */
+    @WrapInTransaction
+    public void doImport() throws Exception {
+
+
+
+        Logger.info(this, "Found " + tempFiles.size() + " files to import");
+
+
+        deleteDotCMS();
+        
+
+        for (File file : endsWith("Company.xml")) {
+            doXMLFileImport(file);
+        }
+
+
+        List<Role> roles = new ArrayList<Role>();
+
+        for (File file : startsWith("com.dotmarketing.business.Role_")) {
+            roles.addAll((List<Role>) BundlerUtil.xmlToObject(file));
+        }
+
+        Collections.sort(roles);
+
+
+        for (Role role : roles) {
+            HibernateUtil _dh = new HibernateUtil(Role.class);
+            String id = HibernateUtil.getSession().getSessionFactory().getClassMetadata(Role.class).getIdentifierPropertyName();
+
+            if(UtilMethods.isSet(role.getRoleKey())) {
+               List<Map<String,Object>>  matches= new DotConnect().setSQL("select * from cms_role where role_key =?").addParam(role.getRoleKey()).loadObjectResults();
+               Logger.info(this.getClass(), "roleKey:" +role.getRoleKey() + " = " + matches.size() );
+            }
+
+            _dh.saveWithPrimaryKey(role, role.getId());
+            HibernateUtil.getSession().flush();
+        }
+
+        for (File file : endsWith(User.class.getCanonicalName() + ".xml")) {
+            doXMLFileImport(file);
+        }
+
+        for (File file : contains(UsersRoles.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(Layout.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+        
+        
+        for (File file : contains(PortletsLayouts.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+
+        
+        for (File file : endsWith("LogsMappers.xml")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(LayoutsRoles.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+        
+        
+        
+        for (File file : contains(Language.class.getCanonicalName() + "_")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'file' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)